dxcint/dxcint/Terraform.py

```diff
@@ -0,0 +1,234 @@
+        return list(assets.values())
+
+    def _clean_up(self, language: str) -> bool:
+        """
+        Method to remove local asset directories for a given language.
+        Args:
+            language: str. Language for which to destroy terraformed paths with resources.
+
+        Returns: bool. True if all is done
+        """
+        language_dir = os.path.join(self._context.repo_root_dir, "applet_resources", language)
+        if os.path.exists(language_dir):
+            shutil.rmtree(language_dir)
+        else:
+            self._context.logger.info(
+                f"Terraform._clean_up(): No local asset directories for `{language}` language were pre-existing"
+            )
+        return True
+
+    def _wdl_asset(self) -> str:
+        return self._make_prerequisites("wdl")
+
+    def _cwl_asset(self) -> str:
+        return self._make_prerequisites("cwl")
+
+    def _make_prerequisites(self, language: str) -> str:
+        always_capital_lang = language.upper()
+        language_specific_dependencies = [x for x in self._dependencies if language in x.languages]
+        try:
+            local_asset_dirs = self._create_local_asset_dir(always_capital_lang)
+            for dependency in language_specific_dependencies:
+                _ = dependency.link(local_asset_dirs.get("bin"))
+            _ = self._create_asset_spec(always_capital_lang)
+            asset_id = self._build_asset(always_capital_lang)
+            _ = self._clean_up(always_capital_lang)
+            return asset_id
+        except Exception as e:
+            _ = self._clean_up(always_capital_lang)
+            raise e
+
+    @async_retry()
+    def _build_asset(self, language: str) -> str:
+        asset_name = f"dx{language}rt"
+        destination = f"{self._context.project_id}:{self._context.platform_build_dir}/{asset_name}"
+        cwd = os.getcwd()
+        os.chdir(os.path.join(self._context.repo_root_dir, "applet_resources"))
+        with self._context.lock:
+            self._context.logger.info(f"Terraform._build_asset(): Creating a runtime asset for {language}")
+        proc = sp.Popen(
+            ["dx", "build_asset", language, "--destination", destination],
+            stdout=sp.PIPE,
+            stderr=sp.PIPE
+        )
+        out, err = proc.communicate()
+        if proc.returncode != 0:
+            raise TerraformError(f"Building DNAnexus asset raised {err.decode()}")
+        os.chdir(cwd)
+        asset_id = dxpy.search.find_one_data_object(
+            classname="record",
+            project=self._context.project_id,
+            name=asset_name,
+            folder=self._context.platform_build_dir,
+            more_ok=False
+        )
+        with self._context.lock:
+            self._context.logger.info(
+                f"Successfully created asset for {language.upper()}. Asset ID:{asset_id.get('id')}"
+            )
+        return asset_id.get("id")
+
+    def _create_asset_spec(self, language: str) -> Dict:
```

Comment:
Will this also be used when we build assets during the release?

Reply:
I aimed to standardize all dependency delivery and asset building as much as possible, so shouldn't be a problem. Can build a special CLI command for that to only terraform for releases.